overlord/snapstate/handlers.go

```diff
@@ -346,28 +355,17 @@ func (m *SnapManager) installOneBaseOrRequired(t *state.Task, snapName string, c
 		}
 
 		// other kind of dependency, check if it's in progress
-		if ok, err := inProgress(snapName); err != nil {
+		if _, err := shouldAttemptInstall(snapName); err != nil {
```

Comment:
do we actually need this code inside the "if" now? wouldn't the code outside the "if" do the same in practice?

Reply:
No, it can be dropped here to just always return `err`, regardless if nil or not.